### PR TITLE
ocrmypdf: 9.8.2 -> 10.2.0

### PR DESCRIPTION
**Pull request by bachp**
###### Motivation for this change

Update to latest version.

`coloredlogs` is required in a newer version, so I updated its too.

###### Things done

@flokli @Mic92 I enabled the tests as discussed in https://github.com/NixOS/nixpkgs/pull/89710

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/ocrmypdf/default.nix [hunk 1]
```diff
@@ -77,23 +79,7 @@ buildPythonApplication rec {
...
-    pytest -k 'not test_force_ocr_on_pdf_with_no_images \
-    and not test_tesseract_crash \
-    and not test_tesseract_crash_autorotate \
-    and not test_ghostscript_pdfa_failure \
-    and not test_gs_render_failure \
-    and not test_gs_raster_failure \
-    and not test_bad_utf8 \
-    and not test_old_unpaper'
-  '';
+  #doCheck = false;
```
Comment (flokli) [inline pkgs/tools/text/ocrmypdf/default.nix:82]: can this line be removed?

**Review by flokli — CHANGES_REQUESTED** [review]

File: pkgs/tools/text/ocrmypdf/default.nix (hunk 1 above)
Comment (bachp) [reply pkgs/tools/text/ocrmypdf/default.nix:82]: I left it in because of the comment.

Comment (eyJhb) [reply pkgs/tools/text/ocrmypdf/default.nix:82]: I had the exact same thought proces as you two just had.

Comment (flokli) [reply pkgs/tools/text/ocrmypdf/default.nix:82]: Yeah, let's remove this line. If this turns out to be a problem, having to write this line is okay :-)

Comment (bachp) [reply pkgs/tools/text/ocrmypdf/default.nix]: Ok removed

**Pull request merged**